File: golang/01/conformity.go
package main

import (
  "fmt"
  "sort"
)


func main() {
  for {
    // Store line number in a variable
    var lines int
    fmt.Scanf("%d\n", &lines)

    if lines == 0 {
      break
    }

    // Map in which key is five int array and value is the number of times the array is found
    var frosh = make(map[[5]int]int)

    for i := 0; i < lines; i++ {
      // Read line with five numbers and store them in array
      var numbers [5]int
      fmt.Scanf("%d %d %d %d %d\n", &numbers[0], &numbers[1], &numbers[2], &numbers[3], &numbers[4])

      // Sort array of integers
      sort.Ints(numbers[:])

      // Increment value in map
      frosh[numbers]++
    }

    // Find the largest value in the map
    largest := 0
    for _, value := range frosh {
      if value > largest {
        largest = value
      }
    }

    // Count number of times the largest value is found in the map
    count := 0
    for _, value := range frosh {
      if value == largest {
        count += value
      }
    }

    // Print number of times the largest value is found
    fmt.Printf("%d\n", count)
  }
}

File: cpp/01/conformity.cpp
#include <iostream>
#include <array>
#include <map>
#include <algorithm>

int main (void) {
  while (true) {
    int lines;
    std::cin >> lines;

    // Exit condition
    if (lines == 0) {
      return 0;
    }

    // Stores a line as a key and the value counts as the number of times the line is repeated
    std::map<std::array<int, 5>, int> frosh;

    // For each line
    for (int i = 0; i < lines; i++) {
      // Capture numbers in array for key
      std::array<int, 5> this_line;
      for (int j = 0; j < 5; j++) {
        std::cin >> this_line[j];
      }

      // Sort the array
      std::sort(this_line.begin(), this_line.end());

      // Increment key
      frosh[this_line]++;
    }

    // Find largest value in map
    int largest = 0;
    for (auto const& entry : frosh) {
      int value = entry.second;
      if (value > largest) {
        largest = value;
      }
    }

    // Count number of keys with largest value
    int count = 0;
    for (auto const& entry : frosh) {
      int value = entry.second;
      if (value == largest) {
        count += value;
      }
    }

    std::cout << count << std::endl;
  }
}
